invoke/tasks.py

```diff
@@ -286,7 +287,7 @@ def get_arguments(
         return args
 
 
-def task(*args: Any, **kwargs: Any) -> Callable:
+def task(*args: Any, **kwargs: Any) -> "Task[T]":
```

Comment:
The TypeVar is not being used correctly in this function. Either use `Task[Any]`, or if you want to spend a bit more time on it, you could try and type it properly.

I think the correct signature would look something along the lines of:
```
@overload
def task(arg: T, *, klass: Callable[[T]], U], **kwargs: Any) -> U:
    ...
@overload
def task(arg: T, **kwargs: Any) -> Task[T]:
    ...
@overload
def task(*args, klass: Callable[[Callable], T], **kwargs: Any) -> Callable[[Callable], T]:
    ...
@overload
def task(*args: Any, **kwargs: Any) -> Callable[[T], Task[T]]:  # T should actually be bound to Callable on this one.
    ...
```

But, frankly, it's rather complicated.

Reply:
This is following the recommendation provided by myotonic documentation: https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

So, while it could be argued that it's not correct, it's still the recommended solution.

Reply:
I think every example I see in that documentation matches my suggestions, not the original code.

The mistake is that a TypeVar is being used as a return type, but does not appear in the parameters (or anywhere else), which means that it is not doing the job of a TypeVar. The point of a TypeVar is to infer the precise type based on context. e.g. `def foo(a: _T) -> _T:` means that the return type will be the same as the input type. If you only put it in one place, then this gives no information at all (it wouldn't surprise me if mypy were to produce a type error in future, due to this being obviously incorrect usage).

Reply:
Seems that mypy already produces an error for this, it's a bug that it didn't trigger for you:
https://github.com/python/mypy/issues/16113